--- CMakeLists1.txt	2023-07-29 20:03:46.335762299 +0800
+++ CMakeLists.txt	2023-07-29 20:04:52.139895541 +0800
@@ -935,18 +935,6 @@ file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/
 # Set CMAKE_INSTALL_LIBDIR and friends.
 include(GNUInstallDirs)
 
-# Install the library binary. The INCLUDES specifies the include path that
-# is exported for other projects to use but it doesn't install any files.
-install(TARGETS liblzma EXPORT liblzmaTargets
-        RUNTIME  DESTINATION "${CMAKE_INSTALL_BINDIR}"
-                 COMPONENT liblzma_Runtime
-        LIBRARY  DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-                 COMPONENT liblzma_Runtime
-                 NAMELINK_COMPONENT liblzma_Development
-        ARCHIVE  DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-                 COMPONENT liblzma_Development
-        INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
-
 # Install the liblzma API headers. These use a subdirectory so
 # this has to be done as a separate step.
 install(DIRECTORY src/liblzma/api/
@@ -959,12 +947,6 @@ set(liblzma_INSTALL_CMAKEDIR
     "${CMAKE_INSTALL_LIBDIR}/cmake/liblzma"
     CACHE STRING "Path to liblzma's .cmake files")
 
-install(EXPORT liblzmaTargets
-        NAMESPACE liblzma::
-        FILE liblzma-targets.cmake
-        DESTINATION "${liblzma_INSTALL_CMAKEDIR}"
-        COMPONENT liblzma_Development)
-
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/liblzma-config.cmake"
               "${CMAKE_CURRENT_BINARY_DIR}/liblzma-config-version.cmake"
         DESTINATION "${liblzma_INSTALL_CMAKEDIR}"
