From 7b833e3e1189ef3744bf8a57994e5155739bbc16 Mon Sep 17 00:00:00 2001
From: affggh <879632264@qq,com>
Date: Sun, 24 Sep 2023 09:56:29 +0800
Subject: [PATCH] Cygwin: libcutils: Add cygwin flag on threads.cpp

---
 libcutils/threads.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libcutils/threads.cpp b/libcutils/threads.cpp
index 6ece7a3af..824f41415 100644
--- a/libcutils/threads.cpp
+++ b/libcutils/threads.cpp
@@ -21,7 +21,7 @@
 #elif defined(__linux__)
 #include <syscall.h>
 #include <unistd.h>
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__CYGWIN__)
 #include <windows.h>
 #endif
 
@@ -36,7 +36,7 @@ pid_t gettid() {
   return tid;
 #elif defined(__linux__)
   return syscall(__NR_gettid);
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__CYGWIN__)
   return GetCurrentThreadId();
 #endif
 }
-- 
2.39.0

