From cfd4bd299bdf6cd9feb2ff0d14059ba294cfb2e4 Mon Sep 17 00:00:00 2001
From: affggh <879632264@qq,com>
Date: Sun, 24 Sep 2023 09:58:48 +0800
Subject: [PATCH] Cygwin: libbase: Add cygwin flags

---
 file.cpp    | 2 +-
 threads.cpp | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/file.cpp b/file.cpp
index e433a07..b019c7f 100644
--- a/file.cpp
+++ b/file.cpp
@@ -468,7 +468,7 @@ bool Realpath(const std::string& path, std::string* result) {
 #endif
 
 std::string GetExecutablePath() {
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
   std::string path;
   android::base::Readlink("/proc/self/exe", &path);
   return path;
diff --git a/threads.cpp b/threads.cpp
index 4197ff8..b5ce7b7 100644
--- a/threads.cpp
+++ b/threads.cpp
@@ -23,7 +23,7 @@
 #include <pthread.h>
 #elif defined(__linux__) && !defined(__ANDROID__)
 #include <syscall.h>
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__CYGWIN__)
 #include <windows.h>
 #endif
 
@@ -39,7 +39,7 @@ uint64_t GetThreadId() {
   return tid;
 #elif defined(__linux__)
   return syscall(__NR_gettid);
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__CYGWIN__)
   return GetCurrentThreadId();
 #endif
 }
-- 
2.39.0

